compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -682,16 +682,304 @@ abstract class GenJSCode extends plugins.PluginComponent
+                  }
+              }.getOrElse(Nil)
+
+              beforeSuperCall ::: superCallParams
+
+            case js.ApplyStatic(cls, method, args) =>
+              zipMap(outputParams, args.tail) { (ref, tree) =>
+                  js.Assign(ref, tree)
+              }
+
+            case _ => Nil
+          }
+        }
+
+        js.Block(subCalls :: preSuperCall)
+      }
+
+      private[ConstructorGraph] def mkSubPostCalls(overrideNumRef: js.VarRef,
+          isConstructor: Set[String])(implicit pos: Position): js.Tree = {
+        val overrideNumss = subConstructors.map(_.overrideNumBounds)
+        val bodies = subConstructors.map { cg =>
+          cg.mkPostSuperCallOnSndCtr(overrideNumRef, isConstructor)
+        }
+        overrideNumss.zip(bodies).foldRight[js.Tree](js.Skip()) {
+          case ((numBounds, js.Skip()), acc) => acc
+
+          case ((numBounds, body), acc) =>
+            val cond = mkOverrideNumsCond(overrideNumRef, numBounds)
+            js.If(cond, body, acc)(jstpe.BooleanType)
+        }
+      }
+
+      private[ConstructorGraph] def mkPostSuperCallOnSndCtr(overrideNumRef: js.VarRef,
+          isConstructor: Set[String])(implicit pos: Position): js.Tree = {
+        val postSuperCall = {
+          method.body match {
+            case js.Block(stats) =>
+              stats.dropWhile {
+                case js.ApplyStatic(_, mtd, _) if isConstructor(mtd.name) => false
+                case _                                                    => true
+              }.tail
+
+            case _ => Nil
+          }
+        }
+        js.Block(postSuperCall :+ mkSubPostCalls(overrideNumRef, isConstructor))
+      }
+
+      private def mkOverrideNumsCond(numRef: js.VarRef,
+          numBounds: (Int, Int)) = numBounds match {
+        case (lo, hi) if lo == hi =>
+          js.BinaryOp(js.BinaryOp.===, js.IntLiteral(lo), numRef)
+
+        case (lo, hi) if lo == hi - 1 =>
+          val lhs = js.BinaryOp(js.BinaryOp.===, numRef, js.IntLiteral(lo))
+          val rhs = js.BinaryOp(js.BinaryOp.===, numRef, js.IntLiteral(hi))
+          js.If(lhs, js.BooleanLiteral(true), rhs)(jstpe.BooleanType)
+
+        case (lo, hi) =>
+          val lhs = js.BinaryOp(js.BinaryOp.Num_<=, js.IntLiteral(lo), numRef)
+          val rhs = js.BinaryOp(js.BinaryOp.Num_<=, numRef, js.IntLiteral(hi))
+          js.BinaryOp(js.BinaryOp.Boolean_&, lhs, rhs)
+          js.If(lhs, rhs, js.BooleanLiteral(false))(jstpe.BooleanType)
+      }
+    }
+
+    private def zipMap[T, U, V](xs: List[T], ys: List[U])(
+        f: (T, U) => V): List[V] = {
+      if (xs.nonEmpty && ys.nonEmpty)
+        f(xs.head, ys.head) :: zipMap(xs.tail, ys.tail)(f)
+      else
+        Nil
     }
 
+    private def mkOverloadSelection(constructorGraph: ConstructorGraph,
+        overloadIdent: js.Ident, dispatchResolution: js.Tree,
+        isConstructor: Set[String])(implicit pos: Position): List[js.Tree] = {
```

Comment:
This needs a good comment explaining what it does, and the general aspect of trees it returns.